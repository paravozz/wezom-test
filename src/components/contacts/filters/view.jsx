import React from 'react';
import { Row, Input, Select, Col, Checkbox, Button } from 'antd';
import { CloseOutlined } from '@ant-design/icons';
import { CountryData } from 'utils';
import { getGenderName } from 'utils';


const View = (props) => {
	const {
		loading,
		filters,
		filterContacts,
	} = props;

	return (
		<div className="box">
			<Row align="middle">
				<Col className="_flex-grow">
					<Row gutter={[6, 6]} align="middle">
						<Col xs={24} lg={11}>
							<Row>
								<Input.Search
									loading={loading}
									placeholder="Search by full name"
									size="large"
									onChange={e => filterContacts({ ...filters, name: e.target.value })}
									value={filters.name}
								/>
							</Row>
						</Col>
						<Col xs={24} sm={10} lg={4}>
							<Row>
								<Select
									id="genderSelect"
									loading={loading}
									style={{ width: '100%' }}
									allowClear
									placeholder="Gender"
									size="large"
									onChange={value => filterContacts({ ...filters, gender: value })}
									value={filters.gender || undefined}
								>
									{['male', 'female', 'indeterminate'].map(k => (
										<Select.Option key={k} value={k}>
											{getGenderName(k)}
										</Select.Option>
									))}
								</Select>
							</Row>
						</Col>
						<Col xs={24} sm={14} lg={5}>
							<Row>
								<Select
									id="natSelect"
									loading={loading}
									style={{ width: '100%' }}
									mode="multiple"
									maxTagCount={2}
									allowClear
									size="large"
									placeholder="Nationality"
									onChange={value => filterContacts({ ...filters, nat: value })}
									value={filters.nat}
								>
									{Object.entries(CountryData).map(([key, data]) => (
										<Select.Option key={key} value={key}>{data.name}</Select.Option>
									))}
								</Select>
							</Row>
						</Col>
						<Col xs={24} sm={6} lg={4}>
							<Row>
								<Checkbox
									disabled={loading}
									onChange={(e) => filterContacts({ ...filters, author: e.target.checked })}
								>
									I am creator
								</Checkbox>
							</Row>
						</Col>
					</Row>
				</Col>
				<Col className="_flex-noshrink">
					<Row>
						<Button
							disabled={loading}
							onClick={() => filterContacts(null)}
							type="link"
						>
							<CloseOutlined /> Clear
						</Button>
					</Row>
				</Col>
			</Row>
		</div>
	);
};


export { View };
