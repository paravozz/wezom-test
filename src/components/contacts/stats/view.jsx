import React from 'react';
import { Typography, Col, Row, Statistic, Descriptions } from 'antd';
import {
	countGenders,
	getGenderPredominateMessage,
	countNationalities,
	CountryData,
} from 'utils';


const View = (props) => {
	const { contacts, loading } = props;

	if (!contacts?.length || loading) {
		return (
			<Row className="box" />
		);
	}

	function renderNationalitiesStats(_contacts) {
		const natCount = countNationalities(_contacts);

		return (
			<Descriptions title="Nationalities">
				{Object.entries(natCount).map(([key, value]) => (
					<Descriptions.Item key={key} label={CountryData[key].name}>
						{value} contacts
					</Descriptions.Item>
				))}
			</Descriptions>
		);
	}

	const genderStats = countGenders(contacts);
	const genderPredominateMessage = getGenderPredominateMessage(genderStats);

	return (
		<div className="box">
			<Typography.Title level={3}>Statistic</Typography.Title>
			<Row gutter={24}>
				<Col className="_flex-noshrink">
					<Statistic title="Collection size" value={contacts.length} />
				</Col>
				<Col className="_flex-noshrink">
					<Row gutter={12}>
						<Col span={6}>
							<Statistic title="Males" value={genderStats.male} />
						</Col>
						<Col span={6}>
							<Statistic title="Females" value={genderStats.female} />
						</Col>
						<Col span={6}>
							<Statistic title="Indeterminate" value={genderStats.indeterminate} />
						</Col>
						<Col span={24}>
							<Typography.Text mark>
								{genderPredominateMessage}
							</Typography.Text>
						</Col>
					</Row>
				</Col>
			</Row>
			<br />
			<Row>
				{renderNationalitiesStats(contacts)}
			</Row>
		</div>
	);
};


export { View };
