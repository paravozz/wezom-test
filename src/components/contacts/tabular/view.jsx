import React from 'react';
import { Table, Typography, Tag, Avatar, Empty } from 'antd';
import { UserOutlined } from '@ant-design/icons';
import moment from 'moment';
import { CountryData, getFullName } from 'utils';
import { Copyable } from 'components/utils';


const View = (props) => {
	const { contacts, loading } = props;

	if (!contacts) {
		return (
			<>
				<Empty image={Empty.PRESENTED_IMAGE_SIMPLE} />
			</>
		)
	}

	function renderFullName({ title, first, last }) {
		return getFullName(title, first, last);
	}

	function renderAvatar(pictureUrl) {
		return (
			<Avatar size={40} icon={<UserOutlined />} src={pictureUrl} />
		);
	}

	function renderBirthday({ date, age }) {
		return (
			<div>
				{moment(date).format('dddd, M/D/YYYY, H:m:s A')}
				<br />
				{age} years
			</div>
		);
	}

	function renderEmail(email) {
		return (
			<Copyable text={email}>
				<Typography.Link ellipsis href={`mailto:${email}`}>
					{email}
				</Typography.Link>
			</Copyable>
		);
	}

	function renderPhone(phone) {
		return (
			<Copyable text={phone}>
				<Typography.Link ellipsis href={`tel:${phone}`}>
					{phone}
				</Typography.Link>
			</Copyable>
		);
	}

	function renderLocation({ country, street, city, state, postcode }) {
		return (
			<Copyable
				text={(
					`[${country}] `
					+ `${street.number} ${street.name}, `
					+ `${city}, ${state} ${postcode}`
				)}
			>
				<Typography.Paragraph ellipsis={{ rows: 3 }}>
					<strong>/{country}/</strong>
					<br />
					{(
						`${street.number}, ${street.name}`
						+ `${city}, ${state} ${postcode}`
					)}
				</Typography.Paragraph>
			</Copyable>
		);
	}

	function renderNationality(nat) {
		return (
			<Tag color={CountryData[nat].color}>{CountryData[nat].name}</Tag>
		)
	}

	const dataSource = (
		contacts
			.map((contact, index) => ({
				key: index,
				avatar: renderAvatar(contact.picture.thumbnail),
				fullName: renderFullName(contact.name),
				birthday: renderBirthday(contact.dob),
				email: renderEmail(contact.email),
				phone: renderPhone(contact.phone),
				location: renderLocation(contact.location),
				nationality: renderNationality(contact.nat),
			}))
	);

	const columns = [
		{
			title: 'Avatar',
			dataIndex: 'avatar',
			key: 'avatar',
			fixed: 'left',
			width: 80,
		},
		{
			title: 'Full name',
			dataIndex: 'fullName',
			key: 'fullName',
			width: 155,
			sorter: (a, b) => a.fullName.length - b.fullName.length,
			sortDirections: ['descend', 'ascend'],
		},
		{
			title: 'Birthday',
			dataIndex: 'birthday',
			key: 'birthday',
			width: 250,
		},
		{
			title: 'Email',
			dataIndex: 'email',
			key: 'email',
			width: 155,
		},
		{
			title: 'Phone',
			dataIndex: 'phone',
			key: 'phone',
			width: 155,
		},
		{
			title: 'Location',
			dataIndex: 'location',
			key: 'location',
			width: 250,
		},
		{
			title: 'Nationality',
			dataIndex: 'nationality',
			key: 'nationality',
			width: 155,
			align: 'right',
		},
	];


	return (
		<Table
			dataSource={dataSource}
			columns={columns}
			scroll={{ x: '100vw' }}
			loading={loading}
			pagination={{ position: ['bottomCenter'] }}
		/>
	);
};


export { View };
