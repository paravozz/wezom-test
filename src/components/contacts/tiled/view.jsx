import React from 'react';
import { Typography, Tag, Row, Col, Card, Divider, Empty, Spin } from 'antd';
import { CountryData, getFullName, paginateCollection } from 'utils';
import { Copyable } from 'components/utils';


function renderFullName({ title, first, last }) {
	return getFullName(title, first, last);
}

function renderAvatar(pictureUrl) {
	return (
		<img src={pictureUrl} alt="contact-cover" />
	);
}

function renderBirthday({ age }) {
	return `${age} years`
}

function renderEmail(email) {
	return (
		<Copyable text={email}>
			<Typography.Link ellipsis href={`mailto:${email}`}>
				{email}
			</Typography.Link>
		</Copyable>
	);
}

function renderPhone(phone) {
	return (
		<Copyable text={phone}>
			<Typography.Link ellipsis href={`tel:${phone}`}>
				{phone}
			</Typography.Link>
		</Copyable>
	);
}

function renderLocation({ country, street, city, state, postcode }) {
	return (
		<Copyable
			text={(
				`[${country}] `
				+ `${street.number} ${street.name}, `
				+ `${city}, ${state} ${postcode}`
			)}
		>
			<Typography.Paragraph ellipsis={{ rows: 3 }}>
				<strong>/{country}/</strong>
				<br />
				{(
					`${street.number}, ${street.name}`
					+ `${city}, ${state} ${postcode}`
				)}
			</Typography.Paragraph>
		</Copyable>
	);
}

function renderNationality(nat) {
	return (
		<Tag color={CountryData[nat].color}>{CountryData[nat].name}</Tag>
	)
}

export function transformContacts(pagination, contacts) {
	let baseCollection = (
		pagination
			? paginateCollection(contacts, pagination.offset, pagination.limit)
			: contacts
	);

	return baseCollection.map((contact, index) => ({
		key: index,
		avatar: renderAvatar(contact.picture.large),
		fullName: renderFullName(contact.name),
		birthday: renderBirthday(contact.dob),
		email: renderEmail(contact.email),
		phone: renderPhone(contact.phone),
		location: renderLocation(contact.location),
		nationality: renderNationality(contact.nat),
	}));
}


const View = (props) => {
	const { contacts, loading, pagination } = props;

	if (!contacts?.length || loading) {
		return (
			<>
				{loading && <Spin />}
				<Empty image={Empty.PRESENTED_IMAGE_SIMPLE} />
			</>
		)
	}

	return (
		<Row gutter={[8, 8]}>
			{transformContacts(pagination, contacts).map(contact => (
				<div key={contact.key} style={{ width: '33.3333%', maxWidth: '33.3333%' }}>
					<Col>
						<Card
							hoverable
							cover={contact.avatar}
						>
							<Card.Meta
								title={(
									<Typography.Text>
										<Typography.Text>{contact.fullName}</Typography.Text>
										&nbsp;
										<Typography.Text type="secondary">
											({contact.birthday})
										</Typography.Text>
									</Typography.Text>
								)}
								description={(
									<>
										{contact.email}
										{contact.phone}
										{contact.location}
										<Divider/>
										{contact.nationality}
									</>
								)}
							/>
						</Card>
					</Col>
				</div>
			))}
		</Row>
	);
};


export { View };
