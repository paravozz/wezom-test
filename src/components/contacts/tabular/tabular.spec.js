/* eslint-disable */
import React from 'react';
import { View as TabularContacts } from './view.jsx';
import { Table, Empty } from 'antd';


const setUp = (props) => shallow(<TabularContacts {...props} />);
const setUpRendered = (props) => render(<TabularContacts {...props} />);

describe('TabularContacts', () => {
	let component;

	beforeEach(() => {
		component = setUp();
	});

	it('should render without crashing', () => {
		component = setUpRendered({ contacts });
		expect(component).toMatchSnapshot();
	});

	it('should render without redux props', () => {
		expect(component)
	});

	it('should contain Empty', () => {
		expect((component.find(Empty))?.length).toBe(1);
	});

	it('should render contacts', () => {
		component = setUp({ contacts });

		expect((component.find(Empty))?.length).toBe(0);
		expect((component.find(Table))?.length).toBe(1);
	});
});
