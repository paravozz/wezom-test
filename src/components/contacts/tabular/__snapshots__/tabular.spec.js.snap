// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabularContacts should render without crashing 1`] = `
<div
  class="ant-table-wrapper"
>
  <div
    class="ant-spin-nested-loading"
  >
    <div
      class="ant-spin-container"
    >
      <div
        class="ant-table ant-table-fixed-column ant-table-scroll-horizontal ant-table-has-fix-left"
      >
        <div
          class="ant-table-container"
        >
          <div
            class="ant-table-content"
            style="overflow-x:auto;overflow-y:hidden"
          >
            <table
              style="width:100vw;min-width:100%;table-layout:fixed"
            >
              <colgroup>
                <col
                  style="width:80px;min-width:80px"
                />
                <col
                  style="width:155px;min-width:155px"
                />
                <col
                  style="width:250px;min-width:250px"
                />
                <col
                  style="width:155px;min-width:155px"
                />
                <col
                  style="width:155px;min-width:155px"
                />
                <col
                  style="width:250px;min-width:250px"
                />
                <col
                  style="width:155px;min-width:155px"
                />
              </colgroup>
              <thead
                class="ant-table-thead"
              >
                <tr>
                  <th
                    class="ant-table-cell ant-table-cell-fix-left ant-table-cell-fix-left-last"
                    style="position:sticky;left:0"
                  >
                    Avatar
                  </th>
                  <th
                    class="ant-table-cell ant-table-column-has-sorters"
                  >
                    <div
                      class="ant-table-column-sorters-with-tooltip"
                    >
                      <div
                        class="ant-table-column-sorters"
                      >
                        <span>
                          Full name
                        </span>
                        <span
                          class="ant-table-column-sorter ant-table-column-sorter-full"
                        >
                          <span
                            class="ant-table-column-sorter-inner"
                          >
                            <span
                              aria-label="caret-up"
                              class="anticon anticon-caret-up ant-table-column-sorter-up"
                              role="img"
                            >
                              <svg
                                aria-hidden="true"
                                class=""
                                data-icon="caret-up"
                                fill="currentColor"
                                focusable="false"
                                height="1em"
                                viewBox="0 0 1024 1024"
                                width="1em"
                              >
                                <path
                                  d="M858.9 689L530.5 308.2c-9.4-10.9-27.5-10.9-37 0L165.1 689c-12.2 14.2-1.2 35 18.5 35h656.8c19.7 0 30.7-20.8 18.5-35z"
                                />
                              </svg>
                            </span>
                            <span
                              aria-label="caret-down"
                              class="anticon anticon-caret-down ant-table-column-sorter-down"
                              role="img"
                            >
                              <svg
                                aria-hidden="true"
                                class=""
                                data-icon="caret-down"
                                fill="currentColor"
                                focusable="false"
                                height="1em"
                                viewBox="0 0 1024 1024"
                                width="1em"
                              >
                                <path
                                  d="M840.4 300H183.6c-19.7 0-30.7 20.8-18.5 35l328.4 380.8c9.4 10.9 27.5 10.9 37 0L858.9 335c12.2-14.2 1.2-35-18.5-35z"
                                />
                              </svg>
                            </span>
                          </span>
                        </span>
                      </div>
                    </div>
                  </th>
                  <th
                    class="ant-table-cell"
                  >
                    Birthday
                  </th>
                  <th
                    class="ant-table-cell"
                  >
                    Email
                  </th>
                  <th
                    class="ant-table-cell"
                  >
                    Phone
                  </th>
                  <th
                    class="ant-table-cell"
                  >
                    Location
                  </th>
                  <th
                    class="ant-table-cell"
                    style="text-align:right"
                  >
                    Nationality
                  </th>
                </tr>
              </thead>
              <tbody
                class="ant-table-tbody"
              >
                <tr
                  aria-hidden="true"
                  class="ant-table-measure-row"
                  style="height:0"
                >
                  <td
                    style="padding:0;border:0;height:0"
                  />
                  <td
                    style="padding:0;border:0;height:0"
                  />
                  <td
                    style="padding:0;border:0;height:0"
                  />
                  <td
                    style="padding:0;border:0;height:0"
                  />
                  <td
                    style="padding:0;border:0;height:0"
                  />
                  <td
                    style="padding:0;border:0;height:0"
                  />
                  <td
                    style="padding:0;border:0;height:0"
                  />
                </tr>
                <tr
                  class="ant-table-row ant-table-row-level-0"
                  data-row-key="0"
                >
                  <td
                    class="ant-table-cell ant-table-cell-fix-left ant-table-cell-fix-left-last"
                    style="position:sticky;left:0"
                  >
                    <span
                      class="ant-avatar ant-avatar-circle ant-avatar-image ant-avatar-icon"
                      style="width:40px;height:40px;line-height:40px;font-size:20px"
                    >
                      <img
                        src="https://randomuser.me/api/portraits/thumb/women/42.jpg"
                      />
                    </span>
                  </td>
                  <td
                    class="ant-table-cell"
                  >
                    Mrs Hazel Chen
                  </td>
                  <td
                    class="ant-table-cell"
                  >
                    <div>
                      Friday, 9/9/1977, 7:45:17 AM
                      <br />
                      43 years
                    </div>
                  </td>
                  <td
                    class="ant-table-cell"
                  >
                    <div
                      class="copyable--container"
                    >
                      <span
                        aria-label="copy"
                        class="anticon anticon-copy"
                        role="img"
                        style="color:#1890ff"
                        tabindex="-1"
                      >
                        <svg
                          aria-hidden="true"
                          class=""
                          data-icon="copy"
                          fill="currentColor"
                          focusable="false"
                          height="1em"
                          viewBox="64 64 896 896"
                          width="1em"
                        >
                          <path
                            d="M832 64H296c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h496v688c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8V96c0-17.7-14.3-32-32-32zM704 192H192c-17.7 0-32 14.3-32 32v530.7c0 8.5 3.4 16.6 9.4 22.6l173.3 173.3c2.2 2.2 4.7 4 7.4 5.5v1.9h4.2c3.5 1.3 7.2 2 11 2H704c17.7 0 32-14.3 32-32V224c0-17.7-14.3-32-32-32zM350 856.2L263.9 770H350v86.2zM664 888H414V746c0-22.1-17.9-40-40-40H232V264h432v624z"
                          />
                        </svg>
                      </span>
                      <a
                        class="ant-typography ant-typography-ellipsis"
                        href="mailto:hazel.chen@example.com"
                      >
                        hazel.chen@example.com
                      </a>
                    </div>
                  </td>
                  <td
                    class="ant-table-cell"
                  >
                    <div
                      class="copyable--container"
                    >
                      <span
                        aria-label="copy"
                        class="anticon anticon-copy"
                        role="img"
                        style="color:#1890ff"
                        tabindex="-1"
                      >
                        <svg
                          aria-hidden="true"
                          class=""
                          data-icon="copy"
                          fill="currentColor"
                          focusable="false"
                          height="1em"
                          viewBox="64 64 896 896"
                          width="1em"
                        >
                          <path
                            d="M832 64H296c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h496v688c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8V96c0-17.7-14.3-32-32-32zM704 192H192c-17.7 0-32 14.3-32 32v530.7c0 8.5 3.4 16.6 9.4 22.6l173.3 173.3c2.2 2.2 4.7 4 7.4 5.5v1.9h4.2c3.5 1.3 7.2 2 11 2H704c17.7 0 32-14.3 32-32V224c0-17.7-14.3-32-32-32zM350 856.2L263.9 770H350v86.2zM664 888H414V746c0-22.1-17.9-40-40-40H232V264h432v624z"
                          />
                        </svg>
                      </span>
                      <a
                        class="ant-typography ant-typography-ellipsis"
                        href="tel:(779)-242-0785"
                      >
                        (779)-242-0785
                      </a>
                    </div>
                  </td>
                  <td
                    class="ant-table-cell"
                  >
                    <div
                      class="copyable--container"
                    >
                      <span
                        aria-label="copy"
                        class="anticon anticon-copy"
                        role="img"
                        style="color:#1890ff"
                        tabindex="-1"
                      >
                        <svg
                          aria-hidden="true"
                          class=""
                          data-icon="copy"
                          fill="currentColor"
                          focusable="false"
                          height="1em"
                          viewBox="64 64 896 896"
                          width="1em"
                        >
                          <path
                            d="M832 64H296c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h496v688c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8V96c0-17.7-14.3-32-32-32zM704 192H192c-17.7 0-32 14.3-32 32v530.7c0 8.5 3.4 16.6 9.4 22.6l173.3 173.3c2.2 2.2 4.7 4 7.4 5.5v1.9h4.2c3.5 1.3 7.2 2 11 2H704c17.7 0 32-14.3 32-32V224c0-17.7-14.3-32-32-32zM350 856.2L263.9 770H350v86.2zM664 888H414V746c0-22.1-17.9-40-40-40H232V264h432v624z"
                          />
                        </svg>
                      </span>
                      <div
                        class="ant-typography ant-typography-ellipsis"
                      >
                        <strong>
                          /New Zealand/
                        </strong>
                        <br />
                        6257, Esmonde RoadTimaru, Wellington 27976
                      </div>
                    </div>
                  </td>
                  <td
                    class="ant-table-cell"
                    style="text-align:right"
                  >
                    <span
                      class="ant-tag ant-tag-has-color"
                      style="background-color:#9e1068"
                    >
                      New Zealander
                    </span>
                  </td>
                </tr>
                <tr
                  class="ant-table-row ant-table-row-level-0"
                  data-row-key="1"
                >
                  <td
                    class="ant-table-cell ant-table-cell-fix-left ant-table-cell-fix-left-last"
                    style="position:sticky;left:0"
                  >
                    <span
                      class="ant-avatar ant-avatar-circle ant-avatar-image ant-avatar-icon"
                      style="width:40px;height:40px;line-height:40px;font-size:20px"
                    >
                      <img
                        src="https://randomuser.me/api/portraits/thumb/men/7.jpg"
                      />
                    </span>
                  </td>
                  <td
                    class="ant-table-cell"
                  >
                    Mr Troy Lucas
                  </td>
                  <td
                    class="ant-table-cell"
                  >
                    <div>
                      Wednesday, 10/22/1958, 13:40:29 PM
                      <br />
                      62 years
                    </div>
                  </td>
                  <td
                    class="ant-table-cell"
                  >
                    <div
                      class="copyable--container"
                    >
                      <span
                        aria-label="copy"
                        class="anticon anticon-copy"
                        role="img"
                        style="color:#1890ff"
                        tabindex="-1"
                      >
                        <svg
                          aria-hidden="true"
                          class=""
                          data-icon="copy"
                          fill="currentColor"
                          focusable="false"
                          height="1em"
                          viewBox="64 64 896 896"
                          width="1em"
                        >
                          <path
                            d="M832 64H296c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h496v688c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8V96c0-17.7-14.3-32-32-32zM704 192H192c-17.7 0-32 14.3-32 32v530.7c0 8.5 3.4 16.6 9.4 22.6l173.3 173.3c2.2 2.2 4.7 4 7.4 5.5v1.9h4.2c3.5 1.3 7.2 2 11 2H704c17.7 0 32-14.3 32-32V224c0-17.7-14.3-32-32-32zM350 856.2L263.9 770H350v86.2zM664 888H414V746c0-22.1-17.9-40-40-40H232V264h432v624z"
                          />
                        </svg>
                      </span>
                      <a
                        class="ant-typography ant-typography-ellipsis"
                        href="mailto:troy.lucas@example.com"
                      >
                        troy.lucas@example.com
                      </a>
                    </div>
                  </td>
                  <td
                    class="ant-table-cell"
                  >
                    <div
                      class="copyable--container"
                    >
                      <span
                        aria-label="copy"
                        class="anticon anticon-copy"
                        role="img"
                        style="color:#1890ff"
                        tabindex="-1"
                      >
                        <svg
                          aria-hidden="true"
                          class=""
                          data-icon="copy"
                          fill="currentColor"
                          focusable="false"
                          height="1em"
                          viewBox="64 64 896 896"
                          width="1em"
                        >
                          <path
                            d="M832 64H296c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h496v688c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8V96c0-17.7-14.3-32-32-32zM704 192H192c-17.7 0-32 14.3-32 32v530.7c0 8.5 3.4 16.6 9.4 22.6l173.3 173.3c2.2 2.2 4.7 4 7.4 5.5v1.9h4.2c3.5 1.3 7.2 2 11 2H704c17.7 0 32-14.3 32-32V224c0-17.7-14.3-32-32-32zM350 856.2L263.9 770H350v86.2zM664 888H414V746c0-22.1-17.9-40-40-40H232V264h432v624z"
                          />
                        </svg>
                      </span>
                      <a
                        class="ant-typography ant-typography-ellipsis"
                        href="tel:07-3482-4477"
                      >
                        07-3482-4477
                      </a>
                    </div>
                  </td>
                  <td
                    class="ant-table-cell"
                  >
                    <div
                      class="copyable--container"
                    >
                      <span
                        aria-label="copy"
                        class="anticon anticon-copy"
                        role="img"
                        style="color:#1890ff"
                        tabindex="-1"
                      >
                        <svg
                          aria-hidden="true"
                          class=""
                          data-icon="copy"
                          fill="currentColor"
                          focusable="false"
                          height="1em"
                          viewBox="64 64 896 896"
                          width="1em"
                        >
                          <path
                            d="M832 64H296c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h496v688c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8V96c0-17.7-14.3-32-32-32zM704 192H192c-17.7 0-32 14.3-32 32v530.7c0 8.5 3.4 16.6 9.4 22.6l173.3 173.3c2.2 2.2 4.7 4 7.4 5.5v1.9h4.2c3.5 1.3 7.2 2 11 2H704c17.7 0 32-14.3 32-32V224c0-17.7-14.3-32-32-32zM350 856.2L263.9 770H350v86.2zM664 888H414V746c0-22.1-17.9-40-40-40H232V264h432v624z"
                          />
                        </svg>
                      </span>
                      <div
                        class="ant-typography ant-typography-ellipsis"
                      >
                        <strong>
                          /Australia/
                        </strong>
                        <br />
                        5314, Poplar DrHobart, South Australia 2836
                      </div>
                    </div>
                  </td>
                  <td
                    class="ant-table-cell"
                    style="text-align:right"
                  >
                    <span
                      class="ant-tag ant-tag-magenta"
                    >
                      Australian
                    </span>
                  </td>
                </tr>
                <tr
                  class="ant-table-row ant-table-row-level-0"
                  data-row-key="2"
                >
                  <td
                    class="ant-table-cell ant-table-cell-fix-left ant-table-cell-fix-left-last"
                    style="position:sticky;left:0"
                  >
                    <span
                      class="ant-avatar ant-avatar-circle ant-avatar-image ant-avatar-icon"
                      style="width:40px;height:40px;line-height:40px;font-size:20px"
                    >
                      <img
                        src="https://randomuser.me/api/portraits/thumb/women/86.jpg"
                      />
                    </span>
                  </td>
                  <td
                    class="ant-table-cell"
                  >
                    Mrs Angela Henry
                  </td>
                  <td
                    class="ant-table-cell"
                  >
                    <div>
                      Friday, 7/5/1968, 8:5:23 AM
                      <br />
                      52 years
                    </div>
                  </td>
                  <td
                    class="ant-table-cell"
                  >
                    <div
                      class="copyable--container"
                    >
                      <span
                        aria-label="copy"
                        class="anticon anticon-copy"
                        role="img"
                        style="color:#1890ff"
                        tabindex="-1"
                      >
                        <svg
                          aria-hidden="true"
                          class=""
                          data-icon="copy"
                          fill="currentColor"
                          focusable="false"
                          height="1em"
                          viewBox="64 64 896 896"
                          width="1em"
                        >
                          <path
                            d="M832 64H296c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h496v688c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8V96c0-17.7-14.3-32-32-32zM704 192H192c-17.7 0-32 14.3-32 32v530.7c0 8.5 3.4 16.6 9.4 22.6l173.3 173.3c2.2 2.2 4.7 4 7.4 5.5v1.9h4.2c3.5 1.3 7.2 2 11 2H704c17.7 0 32-14.3 32-32V224c0-17.7-14.3-32-32-32zM350 856.2L263.9 770H350v86.2zM664 888H414V746c0-22.1-17.9-40-40-40H232V264h432v624z"
                          />
                        </svg>
                      </span>
                      <a
                        class="ant-typography ant-typography-ellipsis"
                        href="mailto:angela.henry@example.com"
                      >
                        angela.henry@example.com
                      </a>
                    </div>
                  </td>
                  <td
                    class="ant-table-cell"
                  >
                    <div
                      class="copyable--container"
                    >
                      <span
                        aria-label="copy"
                        class="anticon anticon-copy"
                        role="img"
                        style="color:#1890ff"
                        tabindex="-1"
                      >
                        <svg
                          aria-hidden="true"
                          class=""
                          data-icon="copy"
                          fill="currentColor"
                          focusable="false"
                          height="1em"
                          viewBox="64 64 896 896"
                          width="1em"
                        >
                          <path
                            d="M832 64H296c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h496v688c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8V96c0-17.7-14.3-32-32-32zM704 192H192c-17.7 0-32 14.3-32 32v530.7c0 8.5 3.4 16.6 9.4 22.6l173.3 173.3c2.2 2.2 4.7 4 7.4 5.5v1.9h4.2c3.5 1.3 7.2 2 11 2H704c17.7 0 32-14.3 32-32V224c0-17.7-14.3-32-32-32zM350 856.2L263.9 770H350v86.2zM664 888H414V746c0-22.1-17.9-40-40-40H232V264h432v624z"
                          />
                        </svg>
                      </span>
                      <a
                        class="ant-typography ant-typography-ellipsis"
                        href="tel:0131 413 2855"
                      >
                        0131 413 2855
                      </a>
                    </div>
                  </td>
                  <td
                    class="ant-table-cell"
                  >
                    <div
                      class="copyable--container"
                    >
                      <span
                        aria-label="copy"
                        class="anticon anticon-copy"
                        role="img"
                        style="color:#1890ff"
                        tabindex="-1"
                      >
                        <svg
                          aria-hidden="true"
                          class=""
                          data-icon="copy"
                          fill="currentColor"
                          focusable="false"
                          height="1em"
                          viewBox="64 64 896 896"
                          width="1em"
                        >
                          <path
                            d="M832 64H296c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h496v688c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8V96c0-17.7-14.3-32-32-32zM704 192H192c-17.7 0-32 14.3-32 32v530.7c0 8.5 3.4 16.6 9.4 22.6l173.3 173.3c2.2 2.2 4.7 4 7.4 5.5v1.9h4.2c3.5 1.3 7.2 2 11 2H704c17.7 0 32-14.3 32-32V224c0-17.7-14.3-32-32-32zM350 856.2L263.9 770H350v86.2zM664 888H414V746c0-22.1-17.9-40-40-40H232V264h432v624z"
                          />
                        </svg>
                      </span>
                      <div
                        class="ant-typography ant-typography-ellipsis"
                      >
                        <strong>
                          /United Kingdom/
                        </strong>
                        <br />
                        6473, Park LaneYork, West Midlands M86 0DJ
                      </div>
                    </div>
                  </td>
                  <td
                    class="ant-table-cell"
                    style="text-align:right"
                  >
                    <span
                      class="ant-tag ant-tag-geekblue"
                    >
                      British
                    </span>
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
        </div>
      </div>
      <ul
        class="ant-pagination ant-table-pagination ant-table-pagination-center"
        unselectable="unselectable"
      >
        <li
          aria-disabled="true"
          class="ant-pagination-prev ant-pagination-disabled"
          title="Previous Page"
        >
          <button
            class="ant-pagination-item-link"
            disabled=""
            tabindex="-1"
            type="button"
          >
            <span
              aria-label="left"
              class="anticon anticon-left"
              role="img"
            >
              <svg
                aria-hidden="true"
                class=""
                data-icon="left"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M724 218.3V141c0-6.7-7.7-10.4-12.9-6.3L260.3 486.8a31.86 31.86 0 000 50.3l450.8 352.1c5.3 4.1 12.9.4 12.9-6.3v-77.3c0-4.9-2.3-9.6-6.1-12.6l-360-281 360-281.1c3.8-3 6.1-7.7 6.1-12.6z"
                />
              </svg>
            </span>
          </button>
        </li>
        <li
          class="ant-pagination-item ant-pagination-item-1 ant-pagination-item-active"
          tabindex="0"
          title="1"
        >
          <a
            rel="nofollow"
          >
            1
          </a>
        </li>
        <li
          aria-disabled="true"
          class="ant-pagination-next ant-pagination-disabled"
          title="Next Page"
        >
          <button
            class="ant-pagination-item-link"
            disabled=""
            tabindex="-1"
            type="button"
          >
            <span
              aria-label="right"
              class="anticon anticon-right"
              role="img"
            >
              <svg
                aria-hidden="true"
                class=""
                data-icon="right"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M765.7 486.8L314.9 134.7A7.97 7.97 0 00302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 000-50.4z"
                />
              </svg>
            </span>
          </button>
        </li>
      </ul>
    </div>
  </div>
</div>
`;
